client/client.go

```diff
@@ -1,71 +1,378 @@
 package client
 
 import (
+	"bytes"
+	"encoding/json"
 	"errors"
 	"io"
+	"io/ioutil"
 	"time"
 
-	"github.com/flynn/go-tuf"
 	"github.com/flynn/go-tuf/data"
+	"github.com/flynn/go-tuf/keys"
+	"github.com/flynn/go-tuf/signed"
+	"github.com/flynn/go-tuf/util"
 )
 
 var (
-	ErrNotFound   = errors.New("tuf: file not found")
-	ErrLatest     = errors.New("tuf: the current version is the latest")
-	ErrWrongSize  = errors.New("tuf: unexpected file size")
-	ErrNoRootKeys = errors.New("tuf: no root keys found in local meta store")
+	ErrNotFound       = errors.New("tuf: file not found")
+	ErrLatest         = errors.New("tuf: the current version is the latest")
+	ErrWrongSize      = errors.New("tuf: unexpected file size")
+	ErrNoRootKeys     = errors.New("tuf: no root keys found in local meta store")
+	ErrChecksumFailed = errors.New("tuf: checksum failed")
 )
 
+// LocalStore is local storage for downloaded top-level metadata.
+type LocalStore interface {
+	// GetMeta returns top-level metadata from local storage. The keys are
+	// in the form `ROLE.json`, with ROLE being a valid top-level role.
+	GetMeta() (map[string]json.RawMessage, error)
+
+	// SetMeta persists the given top-level metadata in local storage, the
+	// name taking the same format as the keys returned by GetMeta.
+	SetMeta(name string, meta json.RawMessage) error
+}
+
+// RemoteStore downloads top-level metadata and target files from a remote
+// repository.
 type RemoteStore interface {
-	Get(name string, size int64) (io.ReadCloser, error)
+	// Get downlaods the given file from remote storage.
+	//
+	// `file` should be relative to the root of remote repository (e.g.
+	// "root.json" or "targets/path/to/target/file.txt")
+	Get(file string) (io.ReadCloser, error)
```

Comment:
This should pass the size so that the `Content-Length` can be checked if provided.